Content.Shared/UpgradeKit/Components/UpgradeKitComponent.cs

```diff
@@ -0,0 +1,22 @@
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Shared.UpgradeKit.Components;
+
+[RegisterComponent, NetworkedComponent]
+[AutoGenerateComponentState]
+public sealed partial class UpgradeKitComponent : Component
```

Comment:
everything about this entire system + component could just be done with a construction graph.